### PR TITLE
[MAINTENANCE] Update `datasource new` workflow to enable `ConfiguredAssetDataConnector` usage with SQL-backed `Datasources`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Update YAML snippets to enable integration with `ConfiguredAssetDataConnectors`

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

File: great_expectations/cli/datasource.py [hunk 1]
```diff
@@ -625,6 +636,10 @@ def credentials_snippet(self) -> str:
 private_key_path = "YOUR_KEY_PATH"  # Path to the private key used for authentication
 private_key_passphrase = ""   # Passphrase for the private key used for authentication (optional -- leave blank for none)"""
 
+        snippet += '''\n
+# A table that you would like to add initially as a Data Asset
+table_name = "YOUR_TABLE_NAME"'''
+
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/datasource.py]: Just marking that this snippet is for `snowflake` only

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM. Thank you for this @cdkini.

File: great_expectations/cli/datasource.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/cli/datasource.py]: Correct!

**Comment by Shinnnyshinshin** [comment]
sorry about the force push. I forgot to do a pull before pushing the changes, and Lazygit is a little trigger-happy with force pushing 😓

**Review by alexsherstinsky — APPROVED** [review]
LGTM (assuming that the failing test will be fixed).

**Review by kenwade4 — APPROVED** [review]

**Pull request merged**